# A baseball player is to play in the World Series

# Based upon his season play, you estimate that if he comes to bat four times in a game the number of hits he will get has a distribution

# $p_{X} = \left\{\begin{matrix}0.4 & ;0\\ 0.2 & ;1\\0.2 & ;2\\0.1 & ;3\\ 0.1 & ;4\end{matrix}\right.$

# Assume a player has four at bats in each game

## a) Let $X$ denote the number of hits that he gets in a series. Using the program *NFoldConvolution*, find the distribution of $X$ for each of the possible series lengths (4, 5, 6, or 7)

## b) Using the distribution found in a), find the probability that his batting average is over 0.400 in a four game series

## c) Given the distribution $p_{X}$, what is his long term batting average?

____

# a)

In [22]:
def P_X(x):
    if x == 0:
        return 0.4
    elif x in [1,2]:
        return 0.2
    elif x in [3,4]:
        return 0.1
    else:
        return 0

In [25]:
def NFoldConvolution(x, n):
    if n == 1:
        return P_X(x)
    elif n == 2:
        total = 0
        for i in range(x+1):
            total += P_X(i)*P_X(x-i)
        return total
    else:
        total = 0
        for i in range(x+1):
            total += NFoldConvolution(x-i, n-1)*P_X(i)
        return total

In [40]:
dict_probabilities = {}

for n_games in [4, 5, 6, 7]:
    probs = []
    for n_hits in range(4*n_games+1):
        val = NFoldConvolution(n_hits, n_games)
        probs.append(val)
    dict_probabilities[n_games] = probs

In [28]:
import pandas as pd

In [41]:
df = pd.DataFrame(index = range(7*4+1))

for n_games in [4,5,6,7]:
    df.loc[:4*n_games, n_games] = dict_probabilities[n_games]

In [42]:
df

4        5         6             7
0   0.0256  0.01024  0.004096  1.638400e-03
1   0.0512  0.02560  0.012288  5.734400e-03
2   0.0896  0.05120  0.027648  1.433600e-02
3   0.1152  0.07680  0.047104  2.723840e-02
4   0.1424  0.10560  0.071424  4.515840e-02
5   0.1408  0.12192  0.092928  6.487040e-02
6   0.1312  0.12960  0.110144  8.440320e-02
7   0.1056  0.12240  0.117504  9.946880e-02
8   0.0808  0.10800  0.116352  1.085056e-01
9   0.0528  0.08560  0.105472  1.092672e-01
10  0.0328  0.06352  0.089328  1.028608e-01
11  0.0176  0.04280  0.069984  9.033920e-02
12  0.0089  0.02700  0.051424  7.461440e-02
13  0.0036  0.01530  0.034992  5.775840e-02
14  0.0014  0.00810  0.022332  4.214720e-02
15  0.0004  0.00381  0.013184  2.887920e-02
16  0.0001  0.00165  0.007272  1.865360e-02
17     NaN  0.00060  0.003672  1.129240e-02
18     NaN  0.00020  0.001721  6.428800e-03
19     NaN  0.00005  0.000726  3.414600e-03
20     NaN  0.00001  0.000279  1.695400e-03
21     NaN      NaN  0.000092  7.771000e-04
22     NaN      NaN  0.000027  3.297000e-04
23     NaN      NaN  0.000006  1.267000e-04
24     NaN      NaN  0.000001  4.410000e-05
25     NaN      NaN       NaN  1.330000e-05
26     NaN      NaN       NaN  3.500000e-06
27     NaN      NaN       NaN  7.000000e-07
28     NaN      NaN       NaN  1.000000e-07

_____

# b)

# If his batting average is above 0.400, he had at least 7 hits in 16 at bats

In [44]:
df.loc[7:, 4].sum()

0.3040000000000001

# So $P(\text{Batting Avg.}\geq0.400) = 0.304$

_____

# c)

# Let $S_{n}$ be the number of hits in $n$ games

# Then $S_{n} = X_{1} + X_{2} + ... + X_{n}$ where each $X_{i}$ represents the number of hits in a single game

# Then $E(S_{n}) = E(X_{1}) + E(X_{2}) + ... + E(X_{n})$

# $E(X_{i}) = (1)(0.2) + (2)(0.2) + (3)(0.1) + (4)(0.1)= 1.3$

# $\implies E(S_{n}) = n(1.3)$

# Then, $E(\text{Batting Avg. in }n\text{ games}) = E(\frac{S_{n}}{4\cdot n}) = E(\frac{1.3}{4}) = 0.325$